modules/core/src/lib/deck-picker.ts

```diff
@@ -319,7 +497,117 @@ export default class DeckPicker {
+    const leftTop = this.device.canvasContext.cssToDevicePixels([x, y], true);
+
+    // take left and top (y inverted in device pixels) from start location
+    const deviceLeft = leftTop.x;
+    const deviceTop = leftTop.y + leftTop.height;
+
+    // take right and bottom (y inverted in device pixels) from end location
+    // @ts-expect-error TODO - assuming WebGL context
+    const rightBottom = this.device.canvasContext.cssToDevicePixels([x + width, y + height], true);
+    const deviceRight = rightBottom.x + rightBottom.width;
+    const deviceBottom = rightBottom.y;
+
+    const deviceRect = {
+      x: deviceLeft,
+      y: deviceBottom,
+      // deviceTop and deviceRight represent the first pixel outside the desired rect
+      width: deviceRight - deviceLeft,
+      height: deviceTop - deviceBottom
+    };
+
+    const pickedResult = this._drawAndSample({
+      layers: pickableLayers,
+      views,
+      viewports,
+      onViewportActive,
+      deviceRect,
+      cullRect: {x, y, width, height},
+      effects,
+      pass: `picking:${mode}`
+    });
+
+    const pickInfos = getUniqueObjects(pickedResult);
+
+    // `getUniqueObjects` dedup by picked color
+    // However different picked color may be linked to the same picked object, e.g. stroke and fill of the same polygon
+    // picked from different sub layers of a GeoJsonLayer
+    // Here after resolving the picked index with `layer.getPickingInfo`, we need to dedup again by unique picked objects
+    const uniquePickedObjects = new Map<string, Set<unknown>>();
+    const uniqueInfos: PickingInfo[] = [];
+
+    const limitMaxObjects = Number.isFinite(maxObjects);
+
+    for (let i = 0; i < pickInfos.length; i++) {
+      if (limitMaxObjects && uniqueInfos.length >= maxObjects!) {
+        break;
+      }
+      const pickInfo = pickInfos[i];
+      let info: PickingInfo = {
+        color: pickInfo.pickedColor,
+        layer: null,
+        index: pickInfo.pickedObjectIndex,
+        picked: true,
+        x,
+        y,
+        pixelRatio
+      };
+
+      info = getLayerPickingInfo({layer: pickInfo.pickedLayer as Layer, info, mode});
+      // info.layer is always populated because it's a picked pixel
+      const pickedLayerId = info.layer!.id;
+      if (!uniquePickedObjects.has(pickedLayerId)) {
+        uniquePickedObjects.set(pickedLayerId, new Set<unknown>());
+      }
+      const uniqueObjectsInLayer = uniquePickedObjects.get(pickedLayerId) as Set<unknown>;
+      // info.object may be null if the layer is using non-iterable data.
+      // Fall back to using index as identifier.
+      const pickedObjectKey = info.object ?? info.index;
+      if (!uniqueObjectsInLayer.has(pickedObjectKey)) {
+        uniqueObjectsInLayer.add(pickedObjectKey);
+        uniqueInfos.push(info);
+      }
+    }
+
+    return uniqueInfos;
+  }
+
+  /**
+   * Pick all objects within the given bounding box
+   * @deprecated WebGL only
```

Comment:
I don't think we should deprecate these until WebGPU is fully (or mostly) implemented.


I'd also want these to include what to use instead